lib/gc/Target/LLVM/XeVM/Target.cpp

```
@@ -144,9 +144,11 @@ SpirSerializer::moduleToObject(llvm::Module &llvmModule) {
     getOperation().emitError() << "Failed translating the module to Binary.";
     return std::nullopt;
   }
-
-  StringRef bin(serializedSPIRVBinary->c_str(),
-                serializedSPIRVBinary->size() + 1);
+  if (serializedSPIRVBinary->size() % 4) {
+    getOperation().emitError() << "SPIRV code size must be a multiple of 4.";
+    return std::nullopt;
+  }
```

Is this a translator's silent failure that can result in this check being true?

---

This is a requirement for the resulting SPIRV to be valid.
The previous `serializedSPIRVBinary->size() + 1` was the source of `Code size must be a multiple of 4 but is X` type of error, hence the check explicitly verifies the resulting code size.

---

> The previous serializedSPIRVBinary->size() + 1 was the source

Was the size calculation incorrect or are there cases when the returned size in not a multiple of 4?